### ISSUE TITLE
Lower log level for health report messages

### ISSUE THREAD

**Issue by milgner** [hlascelles/que-scheduler#326]
# Description

Running the latest version of que-scheduler, our logs are being spammed with messages like

>I, [2022-03-10T17:29:31.767312 #172174]  INFO -- : {"lib":"que","hostname":"-","pid":172174,"thread":66300,"event":"que-scheduler","message":"que-scheduler last ran at 2022-03-10 17:29:31 +0800."}

There are hundreds of messages like that per second. I'd suggest changing the log level from `info` to `debug` here because in regular use they are expected to occur.

# Versions

ruby version: 3.0.3
que version: ruby-3.0 compatibility pre-release branch
que-scheduler version: 4.2.2 (f8315b573a274dd081c1b60e7708246b756b4044)
postgres version: 13.2

**Comment by hlascelles** [comment]
Thank you this report. I'll investigate this, it certainly shouldn't be happening.

As a first guess, I speculate that two workers may have different opinions as to what "now" is, and the scheduler is oscillating and scheduling itself back and forth depending on when it last thought it ran. This of course should never happen, and is a bug if true. All que "now" checks should always use DB clock time.

Will report back.

**Comment by milgner** [comment]
Maybe this will help with debugging:

Que scheduler runs and immediately re-enqueues itself with the same time. It's a different job as you can see from the `id` but `kwargs` for `run_at` and `last_run_time` are the same.

```
[#<Que::ActiveRecord::Model:0x0000000005c89420
  priority: 0,
  run_at: 2022-03-10 13:46:19.361034 UTC,
  id: 253406,
  job_class: "Que::Scheduler::SchedulerJob",
  error_count: 0,
  last_error_message: nil,
  queue: "default",
  last_error_backtrace: nil,
  finished_at: nil,
  expired_at: nil,
  args: [],
  data: {},
  job_schema_version: 2,
  kwargs: {"queue"=>"default", "run_at"=>"2022-03-10 13:47:00 +0000", "last_run_time"=>"2022-03-10T13:46:19+00:00", "job_dictionary"=>["ProductionConditionJob"]}>]
irb(main):028:0> Que::ActiveRecord::Model.all
=> 
[#<Que::ActiveRecord::Model:0x0000000006a00db0
  priority: 0,
  run_at: 2022-03-10 13:46:19.946189 UTC,
  id: 253570,
  job_class: "Que::Scheduler::SchedulerJob",
  error_count: 0,
  last_error_message: nil,
  queue: "default",
  last_error_backtrace: nil,
  finished_at: nil,
  expired_at: nil,
  args: [],
  data: {},
  job_schema_version: 2,
  kwargs: {"queue"=>"default", "run_at"=>"2022-03-10 13:47:00 +0000", "last_run_time"=>"2022-03-10T13:46:19+00:00", "job_dictionary"=>["ProductionConditionJob"]}>]

```

**Comment by daniel-prause** [comment]
> Thank you this report. I'll investigate this, it certainly shouldn't be happening.
> 
> As a first guess, I speculate that two workers may have different opinions as to what "now" is, and the scheduler is oscillating and scheduling itself back and forth depending on when it last thought it ran. This of course should never happen, and is a bug if true. All que "now" checks should always use DB clock time.
> 
> Will report back.

that would be awesome! having hundreds of queue jobs enqueued per second is really putting the machine to a stress test :D

**Comment by daniel-prause** [comment]
maybe this helps in advance:

![image](https://user-images.githubusercontent.com/5479588/157699551-a74e4ea4-7ab6-4aee-9f6d-d7f7fa46e9e6.png)

**Comment by hlascelles** [comment]
Thanks @daniel-prause, looks like the same thing there too. What versions of everything are you running?

**Comment by hlascelles** [comment]
It sounds like the same problem the ActiveJob adapter will have: https://github.com/que-rb/que/pull/319#issuecomment-1059724003

**Comment by milgner** [comment]
> It sounds like the same problem the ActiveJob adapter will have: [que-rb/que#319 (comment)](https://github.com/que-rb/que/pull/319#issuecomment-1059724003)

Good catch! This looks indeed like it's related. Unfortunately our que version is already at from greensync/que#7db75e02e0b9fa3e60ef96239085f82d13a10836 which already contains the linked greensync/que#500f9a5e23f2097df00b1c6965720a8662976786.

I'll try to find out whether changes in the Que part necessitate changes in que-scheduler or whether the fix needs to happen in the Que core only.

**Comment by hlascelles** [comment]
I have updated the que 2.0 branch to cater for passing in job_options as a separate keyword: `change-all-enqueuing-for-que-2-0`

It should also work for que 1.2.0 and above, but only for 2.0.0.beta  and above if using the ActiveJob wrapper.

Let me know if it helps, and I will cut a gem soon.

**Comment by hlascelles** [comment]
I've released 4.3.1 which I hope fixes this: https://rubygems.org/gems/que-scheduler/versions/4.3.1

Let me know if not!
